HC/WrapperGen/WrapperGen.cpp

```diff
@@ -284,21 +289,28 @@ struct StringFinder
       switch(rop) {
         // type function(type1 arg1, type1 arg2, type2 arg3)
         case PARAMETERS:
+          if(i->getTypeName() == "grid_launch_parm_cxx") {
+              //out << "const ";
+          }
           out << i->getTypeNameWithQual() << " " << i->getArgName();
           break;
         // type function(grid_launch_parm lp, type1 arg2, type2 arg3)
         // retain grid_launch_parm for compatibility
         case PARAMETERSKERNEL:
           if(i->getTypeName() == "grid_launch_parm_cxx")
-            out << "grid_launch_parm" << " " << i->getArgName();
+            out << "const grid_launch_parm" << " " << i->getArgName();
           else
             out << i->getTypeNameWithQual() << " " << i->getArgName();
           break;
         // Class::Class(grid_launch_parm _lp, type1 arg2, type2 arg3)
         case PARAMETERSCONSTR:
-          out << i->getTypeNameWithQual() << " ";
-          if(i->getTypeName() == "grid_launch_parm_cxx")
-            out << "_";
+          if(i->getTypeName() == "grid_launch_parm_cxx") {
+            out << "const " ;
+            out << i->getTypeNameWithQual() << " ";
+            out << "x";
```

Comment:
this is to differentiate "_lp" from "__lp", right? now it becomes "_lp" and "x_lp".